From dd47c7c06136d62a7d96100fba43ab866b4fd679 Mon Sep 17 00:00:00 2001
From: Gurchetan Singh <gurchetansingh@google.com>
Date: Fri, 8 May 2026 11:37:25 -0700
Subject: [PATCH] freedreno: fix ignored qualifier

Fixes:

src/freedreno/common/freedreno_dev_info.h:516:1:
  error: 'const' type qualifier on return type has no effect [-Werror,-Wignored-qualifiers]
  516 | const struct fd_dev_info fd_dev_info(const struct fd_dev_id *id);
      | ^~~~~

Reviewed-by: Eric Engestrom <eric@igalia.com>
Reviewed-by: Valentine Burley <valentine.burley@collabora.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/41518>
---
 src/freedreno/common/freedreno_dev_info.c | 2 +-
 src/freedreno/common/freedreno_dev_info.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/freedreno/common/freedreno_dev_info.c b/src/freedreno/common/freedreno_dev_info.c
index 24779e109d01..271eebc0ab2c 100644
--- a/src/freedreno/common/freedreno_dev_info.c
+++ b/src/freedreno/common/freedreno_dev_info.c
@@ -75,7 +75,7 @@ fd_dev_info_raw(const struct fd_dev_id *id)
    return NULL;
 }
 
-const struct fd_dev_info
+struct fd_dev_info
 fd_dev_info(const struct fd_dev_id *id)
 {
    struct fd_dev_info modified = {};
diff --git a/src/freedreno/common/freedreno_dev_info.h b/src/freedreno/common/freedreno_dev_info.h
index a1299a0dc093..7a302f9eb7bf 100644
--- a/src/freedreno/common/freedreno_dev_info.h
+++ b/src/freedreno/common/freedreno_dev_info.h
@@ -513,7 +513,7 @@ fd_dev_is_supported(const struct fd_dev_id *id) {
 }
 
 /* Final dev info with dbg options and everything else applied.  */
-const struct fd_dev_info fd_dev_info(const struct fd_dev_id *id);
+struct fd_dev_info fd_dev_info(const struct fd_dev_id *id);
 
 const struct fd_dev_info *fd_dev_info_raw_by_name(const char *name);
 
-- 
GitLab

